src/Plugin/Network/GoogleAuth.php

```diff
@@ -4,92 +4,158 @@
    *
-   * @param \Drupal\Core\Render\MetadataBubblingUrlGenerator $url_generator
-   *   Used to generate a absolute url for authentication.
+   * @param \Drupal\social_auth\SocialAuthDataHandler $data_handler
+   *   The data handler.
    * @param array $configuration
    *   A configuration array containing information about the plugin instance.
    * @param string $plugin_id
    *   The plugin_id for the plugin instance.
-   * @param mixed $plugin_definition
+   * @param array $plugin_definition
    *   The plugin implementation definition.
-   * @param EntityTypeManagerInterface $entity_type_manager
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
    *   The entity type manager.
    * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
-   *   The configuration factory.
+   *   The configuration factory object.
+   * @param \Drupal\Core\Logger\LoggerChannelFactoryInterface $logger_factory
+   *   The logger factory.
    */
-  public function __construct(MetadataBubblingUrlGenerator $url_generator, array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entity_type_manager, ConfigFactoryInterface $config_factory) {
+  public function __construct(SocialAuthDataHandler $data_handler,
+                              array $configuration,
+                              $plugin_id,
+                              array $plugin_definition,
+                              EntityTypeManagerInterface $entity_type_manager,
+                              ConfigFactoryInterface $config_factory,
+                              LoggerChannelFactoryInterface $logger_factory,
+                              RequestContext $requestContext) {
+
     parent::__construct($configuration, $plugin_id, $plugin_definition, $entity_type_manager, $config_factory);
 
-    $this->urlGenerator = $url_generator;
+    $this->dataHandler = $data_handler;
+    $this->loggerFactory = $logger_factory;
+    $this->requestContext =$requestContext;
   }
 
   /**
-   * {@inheritdoc}
+   * Sets the underlying SDK library.
+   *
+   * @return \League\OAuth2\Client\Provider\Google
+   *   The initialized 3rd party library instance.
+   *
+   * @throws SocialApiException
+   *   If the SDK library does not exist.
    */
-  public function initSdk() {
-    $class_name = '\Google_Client';
+  protected function initSdk() {
+
+    $class_name = '\League\OAuth2\Client\Provider\Google';
     if (!class_exists($class_name)) {
-      throw new SocialApiException(sprintf('The PHP SDK for Google Services could not be found. Class: %s.', $class_name));
+      throw new SocialApiException(sprintf('The Google Library for the league oAuth not found. Class: %s.', $class_name));
     }
-
     /* @var \Drupal\social_auth_google\Settings\GoogleAuthSettings $settings */
     $settings = $this->settings;
+    // Proxy configuration data for outward proxy.
+    $proxyUrl =  Settings::get("http_client_config")["proxy"]["http"];
```

Comment:
Access settings through dependency injection using the [settings service](https://api.drupal.org/api/drupal/core%21core.services.yml/service/settings/8.4.x)